### DISCUSSION TITLE
URL encoding for use in SPO REST

### DISCUSSION THREAD

**Discussion by badmandragon** [pnp/pnpcore#1256] [Ideas] [answered]
We use the AddAsync  function in adding files to SharePoint.
https://github.com/pnp/pnpcore/blob/3eef092e740a565d639ed998826dc3817290f8a0/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/FileCollection.cs#L24

What we noticed were, for content length <= 10 MB, this uses `decodedUrl='{encodedServerFileName}'` in the `FileUpload` method, however, when files are > 10MB, this uses `url='{newFile.Name}'` in the `ChunkedFileUpload` method. 

**FileUpload**
https://github.com/pnp/pnpcore/blob/3eef092e740a565d639ed998826dc3817290f8a0/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/FileCollection.cs#L76-L77

**ChunkedFileUpload**
https://github.com/pnp/pnpcore/blob/3eef092e740a565d639ed998826dc3817290f8a0/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/FileCollection.cs#L110

The behaviour here is different and with `'` being in file names, depending on file size will depends if we receive errors or not.

I was wondering if this is a feature or does the same encoding need to be applied to the `ChunkedFileUpload` method?

**Comment by jansenbe** [comment, marked as answer]
@badmandragon : thanks for making us aware of this, the chunked upload should also use `addusingpath`. I've updated this and this change will be part of the next nightly build.

**Comment by badmandragon** [comment]
Thanks @jansenbe, Do you have a timeline of when this will be released and when it can be used for a production environment?

**Reply by jansenbe** [reply]
@badmandragon : we don't have a timeline yet for the next major release, but feel free to use the nightly in the mean time. The key difference is that it's binaries are not signed.
